File: java solution/disariumnumber.java
import java.util.*;

class Disarium { 	
	public static void main(String[] args) { 
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the number: ");
        int n = sc.nextInt();
        System.out.println(check(n) ? n + 
                    " is a Disarium Number" : n+ " is not a Disarium Number"); 
    } 
    
    public static boolean check(int n) { 
		int count_digits = Integer.toString(n).length(); 
		int sum = 0; 
		int x = n; 
		while (x!=0) { 
			int r = x%10; 
			sum = (int)(sum + Math.pow(r, count_digits--)); 
			x = x/10; 
		} 
		return (sum == n); 
	} 
} 

File: python solution/disariumnumber.py
import math 

def check(n) : 
	count_digits = len(str(n)) 
	total = 0 
	x = n 
	while (x !=0) : 
		r = x % 10
		total = int(total + math.pow(r, count_digits)) 
		count_digits = count_digits - 1
		x = x/10
		
	if(total == n): 
		return 1
	else: 
		return 0
		

n = int(input("Enter the number: "))
if (check(n) == 1) : 
	print(f"{n} is a Disarium Number")
else : 
	print(f"{n} is not a Disarium Number")

